storage-proofs/src/piece_inclusion_proof.rs

```diff
@@ -485,4 +566,84 @@ mod tests {
         assert_eq!(subtree_capacity(14, 16), 2);
         assert_eq!(subtree_capacity(15, 16), 1);
     }
+
+    fn test_pip_casting<H: Hasher>() {
+        let valid_proof_element_bytes = [
+            1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9,
+            0, 1, 2,
+        ];
+        let valid_proof_element = H::Domain::try_from_bytes(&valid_proof_element_bytes).unwrap();
+
+        let empty_pip: PieceInclusionProof<H> = PieceInclusionProof {
+            position: 0,
+            proof_elements: vec![],
+        };
+
+        let empty_pip_bytes: Vec<u8> = empty_pip.into();
+        assert_eq!(empty_pip_bytes, vec![0, 0, 0, 0, 0, 0, 0, 0]);
+
+        let recast_empty_pip: PieceInclusionProof<H> = empty_pip_bytes.into();
+
+        assert_eq!(recast_empty_pip.position, 0);
+        assert_eq!(recast_empty_pip.proof_elements.len(), 0);
+
+        let valid_pip: PieceInclusionProof<H> = PieceInclusionProof {
+            position: 24,
+            proof_elements: vec![valid_proof_element],
+        };
+
+        let valid_pip_bytes: Vec<u8> = valid_pip.into();
+        assert_eq!(
+            valid_pip_bytes,
+            vec![
+                24, 0, 0, 0, 0, 0, 0, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9,
+                0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2
+            ]
+        );
+
+        let recast_valid_pip: PieceInclusionProof<H> = valid_pip_bytes.into();
+
+        assert_eq!(recast_valid_pip.position, 24);
+        assert_eq!(recast_valid_pip.proof_elements.len(), 1);
+
+        let invalid_pip_bytes: Vec<u8> = vec![0, 0, 0];
```

Comment:
TODO: Add comment explaining why `invalid_pip_bytes` is invalid.

Reply:
TODO: Split the invalid PIP serialization/deserialization stuff into a new test.